ui/pages/confirmations/confirm/confirm.stories.tsx

```diff
@@ -1,27 +1,92 @@
 import React from 'react';
 import { Provider } from 'react-redux';
-import configureStore from '../../../store/store';
+import { cloneDeep } from 'lodash';
+import { unapprovedPersonalSignMsg, signatureRequestSIWE } from '../../../../test/data/confirmations/personal_sign';
+import { unapprovedTypedSignMsgV1, unapprovedTypedSignMsgV4, permitSignatureMsg } from '../../../../test/data/confirmations/typed_sign';
 import mockState from '../../../../test/data/mock-state.json';
+import configureStore from '../../../store/store';
 import ConfirmPage from './confirm';
 
-const store = configureStore({
-  confirm: {
-    currentConfirmation: {
-      type: 'personal_sign',
-    },
-  },
-  metamask: {
-    ...mockState.metamask,
-  },
-});
-
 const ConfirmPageStory = {
   title: 'Pages/Confirm/ConfirmPage',
-  decorators: [(story) => <Provider store={store}>{story()}</Provider>],
+  decorators: [(story) => <div style={{ height: '600px' }}>{story()}</div>],
+}
+
+const argsSignature = {
+  msgParams: { ...unapprovedPersonalSignMsg.msgParams },
+}
+
+const argTypesSignature = {
+  msgParams: {
+    control: 'object',
+    description: '(non-param) overrides currentConfirmation.msgParams',
+  },
+}
+
+function SignatureStoryTemplate(args, confirmation) {
+  const mockConfirmation = cloneDeep(confirmation);
+  mockConfirmation.msgParams = args.msgParams;
+
+  const store = configureStore({
+    confirm: {
+      currentConfirmation: mockConfirmation,
+    },
+    metamask: { ...mockState.metamask },
+  });
+
+  return <Provider store={store}><ConfirmPage /></Provider>;
+}
+
+export const PersonalSignStory = (args) => {
+  return SignatureStoryTemplate(args, unapprovedPersonalSignMsg);
 };
 
-export const DefaultStory = (args) => <ConfirmPage {...args} />;
+PersonalSignStory.storyName = 'Personal Sign';
+PersonalSignStory.argTypes = argTypesSignature;
+PersonalSignStory.args = argsSignature;
 
-DefaultStory.storyName = 'Default';
+export const SignInWithEthereumSIWEStory = (args) => {
+  return SignatureStoryTemplate(args, signatureRequestSIWE);
+};
+
+SignInWithEthereumSIWEStory.storyName = 'Sign-in With Ethereum (SIWE)';
+SignInWithEthereumSIWEStory.argTypes = argTypesSignature;
+SignInWithEthereumSIWEStory.args = {
+  ...argsSignature,
+  msgParams: signatureRequestSIWE.msgParams,
+};
+
+export const SignTypedDataStory = (args) => {
+  return SignatureStoryTemplate(args, unapprovedTypedSignMsgV1);
+};
+
+SignTypedDataStory.storyName = 'SignTypedData';
+SignTypedDataStory.argTypes = argTypesSignature;
+SignTypedDataStory.args = {
+  ...argsSignature,
+  msgParams: unapprovedTypedSignMsgV1.msgParams,
+};
+
+export const PermitStory = (args) => {
+  return SignatureStoryTemplate(args, permitSignatureMsg);
+};
+
+PermitStory.storyName = 'SignTypedData Permit';
+PermitStory.argTypes = argTypesSignature;
+PermitStory.args = {
+  ...argsSignature,
+  msgParams: permitSignatureMsg.msgParams,
+};
+
+export const SignTypedDataV4Story = (args) => {
```

Comment:
Probably a future concern, but assuming we create stories for all the transaction types, is this file likely going to become just the signature stories, so we can scale without one huge file?

Reply:
agreed. It might be a new pattern to create multiple `**.stories.*` file for a single component. added a comment to address this when we extend the story [8510dfc](https://github.com/MetaMask/metamask-extension/pull/25054/commits/8510dfc884dc9ef109d322e8d2d827a0fd516278)